### ISSUE TITLE
Move font shadow settings to FontVariation for consistency with font outlines

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot-proposals#4909]
### Describe the project you are working on

*Related to https://github.com/godotengine/godot-proposals/issues/1963 and https://github.com/godotengine/godot-proposals/issues/2564.*

The Godot editor :slightly_smiling_face: 

### Describe the problem or limitation you are having in your project

- Font outline widths and colors are already defined on a per-font basis, but the shadow outline and color isn't. This creates an inconsistency in the UI design workflow.
- Only a handful of Control-derived nodes support font shadows. Custom drawing doesn't support font shadows unless you do it manually by calling twice.
- Label3D doesn't support font shadows either (only outlines).

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Moving font shadow settings to FontVariation allows using font shadows on more nodes than just Label (and perhaps RichTextLabel), such as Button or ItemList.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- Add font shadow color and X/Y offset properties to FontVariation.
- Remove the font shadow theme items in Label, as a custom font variation can be used instead.
- Add Font shadow Color and Font Shadow Offset properties to Label3D.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

This is about fixing an inconsistency in the UI design workflow.

**Comment by Shadowblitz16** [comment]
what what about ui state does this mean we need diffrent fonts for things like pressed, hovered, disabled, focused?

**Comment by Shadowblitz16** [comment]
https://github.com/godotengine/godot-proposals/issues/2498

**Comment by Shadowblitz16** [comment]
Is this getting added soon I find it a very big limitation.

**Comment by Calinou** [comment]
> Is this getting added soon? I find it a very big limitation.

This proposal breaks compatibility, so it can't be implemented until Godot 5.0 unless we find a way to do it in a 100% backwards-compatible manner.

**Comment by Shadowblitz16** [comment]
> > Is this getting added soon? I find it a very big limitation.
> 
> This proposal breaks compatibility, so it can't be implemented until Godot 5.0 unless we find a way to do it in a 100% backwards-compatible manner.

Why not just add the extra options to font's?

**Comment by Zireael07** [comment]
@Shadowblitz16 this proposal is about *moving* stuff not just adding

**Comment by Etherkin** [comment]
So from my understanding reading the current proposal and the discussion in #9745, for the fix in Godot 4.X, the settings in Label would remain as deprecated, and new properties that do not break compatibility would be added to FontVariation? So this would allow for multiple shadows?

If my understanding is correct, I look forward to this proposal being implemented.

**Comment by Calinou** [comment]
> So this would allow for multiple shadows?

Multiple outlines and shadows are supported in 4.5 using a different, backwards-compatible approach:

- https://github.com/godotengine/godot/pull/104731

**Comment by Etherkin** [comment]
My bad, I realise I did not phrase my question correctly. I'm aware of the new feature, but as far as I know it only applies to Labels. I was wondering if this proposal, if implemented, would work the same way for Buttons.

**Comment by NuclearPhoenixx** [comment]
> My bad, I realise I did not phrase my question correctly. I'm aware of the new feature, but as far as I know it only applies to Labels. I was wondering if this proposal, if implemented, would work the same way for Buttons.

Yeah I hoped so too. It's weird that you can set a font outline, but not the shadow.

**Comment by mik1st** [comment]
Hi, 

I'd just like to show support for this development. In my project, I'm opting to use labels as opposed to more proper control nodes to get the shadow effect.

**Comment by danbrrry** [comment]
Also here to bump this; would really appreciate having an official way to add text shadow onto a button rather than having to resort to some roundabout method of doing so.

**Comment by djrain** [comment]
I'm confused on where this would leave the powerful stacked outlines and shadows feature? It's strange and limited that only Label has this ability. Surely the ideal thing is to move/add the stacked options to FontVariation?
